wordpress-mobile/WordPress-iOS
# Initial use of MySiteViewController

This PR introduces `MySiteViewController` as an alternative to `BlogListViewController` / `BlogDetailViewController` when using the `newNavBarAppearance` feature flag. A few things to note:

- This PR is mainly concerned with testing the My Site section of the app with the feature flag _disabled_, which is now the default.
- As such, there are some known issues when the feature flag is enabled, as the new site picker isn't yet in place. This means that with the feature flag enabled, you are currently unable to switch sites. This will be coming in the next PR.
- I ran into an issue with state restoration once during my testing, where the feature-flagged MySitesViewController was incorrectly restoring a BlogListViewController into the navigation stack before it. If this proves to be a problem for you too, we can address it in a future PR.
- Also note that while `MySiteViewController` contains a lot of code, it is essentially identical to @diegoreymendez's earlier code, which has already been reviewed.

|    |   |
|---|---|
|  ![IMG_0582](https://user-images.githubusercontent.com/4780/110529855-a786f980-8111-11eb-9762-e668fea4931b.PNG) |  ![IMG_0579](https://user-images.githubusercontent.com/4780/110529867-abb31700-8111-11eb-9b21-b65cd897eaa2.PNG) |
| ![IMG_BE1374C89D0B-1](https://user-images.githubusercontent.com/4780/110530223-149a8f00-8112-11eb-8a8e-e6ab4545e3b8.jpeg) | ![IMG_0583](https://user-images.githubusercontent.com/4780/110530141-00ef2880-8112-11eb-90b5-882d776c5b01.PNG) |


**To test:**

- Build and run, and do a smoke test to ensure that the My Site section looks and works as expected on iPhone and iPad.
- Main things to test include logging in and out, logging into an account with no sites and then adding one, and hiding all sites in an account to ensure the no results view is shown.
- Ensure that there is a detail view present when logging into to an account with sites, and that you see a blank (but not grey) detail pane for an account with no sites.
- Also on iPad you can test entering and exiting multitasking mode.
- You can also do a cursory test with the feature flag enabled. Main functionality I've mentioned above should all work, with the exception of switching sites.

PR submission checklist:
